aspnetcore/blazor/webassembly-lazy-load-assemblies.md

```diff
@@ -596,13 +596,15 @@ The assembly is assigned to <xref:Microsoft.AspNetCore.Components.Routing.Router
 
 @code {
     private List<Assembly> lazyLoadedAssemblies = new();
+    private bool grantImaharaRobotControlsAssemblyLoaded;
 
     private async Task OnNavigateAsync(NavigationContext args)
     {
         try
         {
-            if (args.Path == "robot")
+            if ((args.Path == "robot") && !grantImaharaRobotControlsAssemblyLoaded)
             {
+                grantImaharaRobotControlsAssemblyLoaded = true;
```

Comment:
NIT question on the location of this line ...

Because the component survives an exception on the next two lines, does it make sense to move this line (`grantImaharaRobotControlsAssemblyLoaded = true;`) to the end after the `lazyLoadedAssemblies.AddRange(assemblies);` line (i.e., it's only `true` if we know those two lines executed)?

Reply:
I'm going to go ahead and move the line down to get this in. I'm about to shoot a bunch of PRs into the live branch.

Reply:
@guardrex Neither of the two alternatives is perfect. I usually prefer to set the flag before an awaited operation - because with the `await` there, another navigation might occur while waiting for the assembly to load, leaving the flag still `false` for the second run.  

In this case, since the awaited assembly load won’t cause a critical failure if called repeatedly, I agree that your version is better.